### ISSUE TITLE
MacOS supporting

### ISSUE THREAD

**Issue by AlexanderBogatko** [NVIDIA/DALI#2572]
Hi guys! I'm very impressed with the framework, especially the speed of its work, can you please tell me if you have any plans to add macos supporting?

**Comment by Wilann** [comment]
MacOS doesn't support Nvidia GPUs, so likely not coming anytime soon.

**Comment by AlexanderBogatko** [comment]
Thanks for answer, what about a CPU version?

**Comment by Wilann** [comment]
Check out OpenCV's VideoCapture, which works on CPU (currently what I use until I figure out how to use DALI). PyTorch also has a VideoReader, which I think is also for CPU.

**Comment by jantonguirao** [comment]
You can still use DALI without a GPU. When creating a DALI pipeline, just specify `device_id=None`, and that will result in a CPU-only pipeline. Of course, you will not get the benefits of offloading data preprocessing to the GPU, but you can still see performance improvements by using DALI in CPU-only mode.

Regarding MacOS support, it is not in our near future plans to support it, but you could try to make a build yourself, which most likely will require some changes in the codebase and the build system to make it possible.

**Comment by AlexanderBogatko** [comment]
Yes, i can use DALI without GPU on Linux64, but i think there are a lot of people who want to do something like this: install the DALI package (maybe CPU only version) on any (Mac / Windows / Linux) machine (what you prefere), debug augmentation pipeline on this CPU version (locally) and then run it on the NVIDIA GPU server (with Linux64) to boost training process. But it's up to you, guys, maybe there really are external people who want to port the framework. Thanks for answers.

**Comment by jantonguirao** [comment]
I agree that is a fair request. However, due to other features with higher priority, it is unlikely that we can dedicate the time to enable and maintain a MacOS build at the moment.
